webapp/src/app/api/pull-request/[projectName]/route.ts

```diff
@@ -62,19 +62,21 @@ export async function POST(
     const projectStore = await Cache.getProjectStore(projectConfig);
     const languages = await projectStore.getLanguageData();
     const pathsToAdd: string[] = [];
-    // TODO: use forEach and Promise.all
-    for (const lang of Object.keys(languages)) {
-      const yamlPath = path.join(
-        projectConfig.absTranslationsPath,
-        `${lang}.yml`,
-      );
-      const yamlOutput = stringify(unflatten(languages[lang]), {
-        doubleQuotedAsJSON: true,
-        singleQuote: true,
-      });
-      pathsToAdd.push(yamlPath);
-      await fs.writeFile(yamlPath, yamlOutput);
-    }
+    await Promise.all(
```

Comment:
reject is a desirable here, `allSettled` does not throw error, however it is correct that syncLock will unlock before all writing is done which is wrong.
So It is better to use allSettled and check the result then throw error manually in case there was any writeFile error.
for that I a better refactor needed maybe with unit test as well, will do it later in this PR